src/data_retrieval/cds/era5.py

```diff
@@ -20,14 +21,26 @@ def get_era5_cds_metadata(
 
 
 def download_era5_cds(
-    years: List[int], months: List[int], download_path: str, file_name: str
-):
-    retrieve_name: str = "reanalysis-era5-single-levels-monthly-means"
-    file_path: str = os.path.join(download_path, file_name)
-    file_type: str = file_name.split(".")[-1]
-
-    era5_cds_metadata: Dict[str, Any] = get_era5_cds_metadata(
-        years, months, file_type
-    )
-
-    download_cds(retrieve_name, era5_cds_metadata, file_path)
+    years: List[int],
+    months: List[int],
+    file_name: str,
+    download_path: Optional[str] = None,
+) -> Optional[BytesIO]:
+    retrieve_name = "reanalysis-era5-single-levels-monthly-means"
+    file_type = file_name.split(".")[
+        -1
+    ]  # Assumes file_name includes an extension
+
+    # Prepare metadata with appropriate file type
+    era5_cds_metadata = get_era5_cds_metadata(years, months, file_type)
+
+    if download_path:
+        # Construct the full file path if a download path is specified
+        file_path = os.path.join(download_path, file_name)
+        download_cds(retrieve_name, era5_cds_metadata, file_path)
+        print(f"Downloaded: {file_path}")
```

Comment:
If you use `-> Optional[BytesIO]:` in your function return signature, you are saying that the function will return either `BytesIO` object or `None`. If you return nothing in the case of local download, the function return may not be seen as `None` but `undefined`.  That is a bad practice and can break your functionality once your code becomes more complex

```python
return None
```